management/backup.py

```diff
@@ -485,25 +489,42 @@ def list_target_files(config):
 
 		# connect to the region & bucket
 		try:
-			s3 = boto3.client('s3', \
-				endpoint_url=f'https://{target.hostname}', \
-				aws_access_key_id=config['target_user'], \
-				aws_secret_access_key=config['target_pass'])
+			s3 = None
+			if "s3_region_name" in config:
+				s3 = boto3.client('s3', \
+					endpoint_url=config["s3_endpoint_url"], \
+					region_name=config["s3_region_name"], \
+					aws_access_key_id=config["s3_access_key_id"], \
+					aws_secret_access_key=config["s3_secret_access_key"])
+			else:
+				s3 = boto3.client('s3', \
+					endpoint_url=config["s3_endpoint_url"], \
+					aws_access_key_id=config["s3_access_key_id"], \
+					aws_secret_access_key=config["s3_secret_access_key"])
+
 			bucket_objects = s3.list_objects_v2(Bucket=bucket, Prefix=path)['Contents']
```

Comment:
Unrelated to this MR but `['Contents']` will not exist and throw a KeyError when pointed at an empty bucket, which is not helpful when setting up backups for the first time, good opportunity here to fix it imo

Reply:
As soon as I have solved all your other reviews, I will also take care of this.

Reply:
Solved with 928ca81397d73cb552685a5df29af9cec9f5b7c8.